scripts/merge-and-build.sh

```diff
@@ -84,17 +84,29 @@ git clone git@github.com:openshift/ose.git
 cd ose
 if [ "${OSE_VERSION}" == "${OSE_MASTER}" ] ; then
 ## Remove git checkout once master is 3.6
-  git checkout -q enterprise-${OSE_VERSION}
+  git checkout -q fake-master
   git remote add upstream git@github.com:openshift/origin.git --no-tags
   git fetch --all
 
+  # TODO: Once we hard-reset master to contain Origin commits, the carry commits, and the latest tito tag commit
+  # all previous tags are going to be dropped. During the migration, I think we will need to manually tag the
+  # last tito commit or maybe not.
+  last_tag="$( git describe --abbrev=0 --tags )"
+  # Assumes the two latest commits are the tito diffs - won't work otherwise
```

Comment:
Since we're using `tito` to generate one of these commits and the comment two lines down to generate the other, we can do a sanity check here:

```sh
last_commit_subject="$( git log HEAD~2..HEAD~1 --pretty=%s )"
if [[ ! "${last_commit_subject}" ~= Automatic commit of package \[atomic\-openshift\] release \[[0-9]+\.[0-9]+\.[0-9]+\.[0-9]+\.\]\. ]]; then
    echo "[FATAL] The last commit doesn't look like a commit from \`tito\`!"
    echo "[FATAL]   ${last_commit_subject}"
    exit 1
fi

squash_commit_subject="$( git log HEAD~1..HEAD --pretty=%s )"
if [[ "${squash_commit_subject}" != "[CARRY][BUILD] Specfile updates" ]]; then
    echo "[FATAL] The second to last commit doesn't look like a squashed specfile commit!"
    echo "[FATAL]   ${squash_commit_subject}"
    exit 1
fi
```